ONSdigital/dp-cli
# Update controller generation

### What

- Create service layer
- Update to Go 1.17
- Upgrade log.go to v2
- Update CI scripts

### How to review

1. `make build`
2. Create a hello world controller:
`./dp generate-project --create-repository n --name "dp-hello-world-controller" --description "Example controller project generated by dp-cli" --go-version 1.17.1 --port 8124 --project-location . --type controller`
3. Check the generated project is identical to [dp-hello-world-controller](https://github.com/ONSdigital/dp-hello-world/tree/master/dp-hello-world-controller)

### Who can review

Anyone not me
